src/community/wfs-templating/src/main/java/org/geoserver/wfstemplating/response/JSONLDGetFeatureResponse.java

```diff
@@ -0,0 +1,115 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wfstemplating.response;
+
+import java.io.*;
+import java.util.List;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.config.GeoServer;
+import org.geoserver.platform.Operation;
+import org.geoserver.platform.ServiceException;
+import org.geoserver.wfs.WFSGetFeatureOutputFormat;
+import org.geoserver.wfs.request.FeatureCollectionResponse;
+import org.geoserver.wfs.request.GetFeatureRequest;
+import org.geoserver.wfstemplating.builders.TemplateBuilder;
+import org.geoserver.wfstemplating.builders.impl.RootBuilder;
+import org.geoserver.wfstemplating.builders.impl.TemplateBuilderContext;
+import org.geoserver.wfstemplating.builders.jsonld.JsonLdRootBuilder;
+import org.geoserver.wfstemplating.configuration.TemplateConfiguration;
+import org.geoserver.wfstemplating.configuration.TemplateIdentifier;
+import org.geoserver.wfstemplating.writers.JsonLdWriter;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.feature.FeatureIterator;
+
+/**
+ * Encodes features in json-ld output format by means of a ${@link TemplateBuilder} tree obtained by
+ * a JSON-LD template
+ */
+public class JSONLDGetFeatureResponse extends WFSGetFeatureOutputFormat {
+
+    /** The MIME type for a JSON-LD response* */
+    public static final String MIME = "application/ld+json";
+
+    private TemplateConfiguration configuration;
+
+    public JSONLDGetFeatureResponse(GeoServer gs, TemplateConfiguration configuration) {
+        super(gs, MIME);
+        this.configuration = configuration;
+    }
+
+    @Override
+    protected void write(
+            FeatureCollectionResponse featureCollection, OutputStream output, Operation getFeature)
+            throws ServiceException {
+
+        TemplateGetFeatureResponseHelper helper =
+                new TemplateGetFeatureResponseHelper(gs.getCatalog(), TemplateIdentifier.JSONLD);
+        FeatureTypeInfo info =
+                helper.getFeatureTypeInfo(GetFeatureRequest.adapt(getFeature.getParameters()[0]));
+        JsonLdWriter writer = null;
+        try {
+            JsonLdRootBuilder rootBuilder =
```

Comment:
Indeed that is a dummy check.
 The reason for that is that json-ld output is working for one featureType only. To make it able to encode more then one layer a strategy to handle multiple context is needed. Since GeoJSON doesn't have context I made json-ld response encoding multiple layer too, with a temporary workaround using only the context available from the first featureType. 
But yes was a bad idea, since was not a really solution. I've restored the previous behaviour.